### PR TITLE
fix edge case with padding in VelocyPack arrays

### PR DESCRIPTION
**Pull request by jsteemann**
### Scope & Purpose

Fix edge case in VelocyPack when the padding of a 1-byte offsetSize array is removed but the first few entries of the array contain a Slice of type None.

- [ ] Bug-Fix for *devel-branch* (i.e. no need for backports?)
- [x] Bug-Fix for a *released version* (did you remember to port this to all relevant release branches?)
- [x] The behavior in this PR can be (and was) *manually tested* (support / qa / customers can test it)
- [x] The behavior change can be verified via automatic tests

### Testing & Verification

- [x] There are tests in an external testing repository: arangodb/velocypack
- [x] I ensured this code runs with ASan / TSan or other static verification tools

http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/8076/

### CODE REVIEW

**Comment by jsteemann** [comment]
Jenkins tests show an unrelated failure on Windows (authentication tests)

File: CHANGELOG [hunk 1]
```diff
@@ -1,6 +1,9 @@
 v3.6.1 (2020-XX-XX)
 -------------------
 
+* Fixed an edge case in VelocyPack when the padding of a 1-byte offsetSize array 
+  is removed but the first few entries of the array contain a Slice of type None.
```
Comment (mchacki) [inline CHANGELOG]: Should we add that this is not reachable?

Comment (jsteemann) [reply CHANGELOG]: I would leave it as it is, as in theory something else could produce VelocyPack None values.
I don't know what exactly could trigger this, but in theory it could have happened. At least it happened to us when setting the profiling level to that undocumented value. :smile:
